includes/class-cli.php

```diff
@@ -0,0 +1,135 @@
+	 *
+	 * [--code-type=<code_type>]
+	 * : Type of code to generate.
+	 * ---
+	 * default: hash
+	 * options:
+	 *   - hash
+	 *   - letters
+	 *   - number
+	 * ---
+	 *
+	 * [--code-limit=<code_limit>]
+	 * : Number of characters to use for the code.
+	 * ---
+	 * default: 10
+	 *
+	 * [--amount=<amount>]
+	 * : Amount for the discount. This is combined with amount-type.
+	 *
+	 * [--amount-type=<amount_type>]
+	 * : Amount type for the discount.
+	 * ---
+	 * default: percent
+	 * options:
+	 *   - percent
+	 *   - flat
+	 * ---
+	 *
+	 * [--products=<product_ids>]
+	 * : Comma-separated list of product IDs that the discounts will apply to.
+	 *
+	 * [--product-condition=<condition>]
+	 * : Condition for discount to apply to the products.
+	 * ---
+	 * default: any
+	 * options:
+	 *    - all
+	 *    - any
+	 *
+	 * [--global]
+	 * : If not set and products are specified, discounts will only apply to selected products. If set, discounts will
+	 * apply to the entire cart.
+	 *
+	 * [--start=<start>]
+	 * : The start date for the discount code. If omitted, the discount can be used on or after today.
+	 *
+	 * [--expiration=<expiration>]
+	 * : Expiration date of the discounts. If omitted, discounts never expire.
+	 *
+	 * [--min-price=<min_price>]
+	 * : Minimum charge amount before the discount can apply.
+	 *
+	 * [--max-uses=<max_uses>]
+	 * : The maximum number of times this discount can be used. Omit for unlimited.
+	 *
+	 * [--once]
+	 * : Limits the discount to a single-use per customer.
+	 *
+	 * @param array $args
+	 * @param array $assoc_args
+	 */
+	public function __invoke( $args, $assoc_args ) {
+		$final_args = $assoc_args;
+
+		if ( ! isset( $args[0] ) ) {
+			WP_CLI::error( esc_html__( 'A discount name is required.', 'edd_dcg' ) );
+		}
+
+		$final_args['name'] = $args[0];
+
+		// We need to convert the format of a few args.
+		$fields_to_convert = array(
+			'number'            => 'number-codes',
+			'min-price'         => 'min_price',
+			'max-uses'          => 'max',
+			'once'              => 'use_once',
+			'product-condition' => 'product_condition',
+		);
+		foreach ( $fields_to_convert as $cli_key => $function_key ) {
+			if ( array_key_exists( $cli_key, $assoc_args ) ) {
+				$final_args[ $function_key ] = $assoc_args[ $cli_key ];
+				unset( $final_args[ $cli_key ] );
+			}
+		}
+
+		// Convert `products` to an array.
+		if ( ! empty( $final_args['products'] ) ) {
+			$final_args['products']   = array_map( 'intval', explode( ',', $final_args['products'] ) );
+			$final_args['not_global'] = empty( $final_args['global'] );
+
+			unset( $final_args['global'] );
+		}
+
+		WP_CLI::line( esc_html__( 'Creating discount codes...', 'edd_dcg' ) );
+
+		$result = edd_dcg_create_discount_codes( $final_args );
+
+		if ( is_wp_error( $result ) ) {
+			WP_CLI::error( esc_html( $result->get_error_message() ) );
+		} elseif ( true === $result ) {
+			WP_CLI::success( esc_html__( 'Discounts successfully created.', 'edd_dcg' ) );
```

Comment:
```suggestion
			WP_CLI::success( esc_html__( 'Discount codes successfully created.', 'edd_dcg' ) );
```
I was thinking to make it more consistent with the UI (would be nice if they said the same thing), but that one just says "45 codes generated." ... I like that this says successfully, so maybe we could update the UI to say that as well. Made a comment that it would be nice if we could send the number of codes generated back to this message so that we could really achieve parity.